src/block/block_ext.cxx

```diff
@@ -1182,13 +1182,13 @@ __wt_block_extlist_write(
     size = ((size_t)entries + 2) * 2 * WT_INTPACK64_MAXSIZE;
     WT_RET(__wt_block_write_size(session, block, &size));
     WT_RET(__wt_scr_alloc(session, size, &tmp));
-    dsk = tmp->mem;
+    dsk = static_cast<WT_PAGE_HEADER *>(tmp->mem);
```

Comment:
There are more casts because C++ doesn't implicitly coerce pointers to and from `void*`.